activesupport/test/message_verifier_test.rb

```diff
@@ -24,6 +24,7 @@ def test_valid_message
     data, hash = @verifier.generate(@data).split("--")
     assert !@verifier.valid_message?(nil)
     assert !@verifier.valid_message?("")
+    assert !@verifier.valid_message?("\xff")
```

Comment:
Can we throw a `# invalid encoding` on the end of this line, please?

The rest seem self-explanatory as to why they're here, but at a glance, it may not be obvious how this differs from `"purejunk"`.